File: backend/src/main/java/com/richstone/cargo/dto/TopicDto.java
package com.richstone.cargo.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TopicDto {
    @Schema(description = "ID темы", example = "2")
    private Long id;

    @Schema(description = "Название темы", example = "Geography")
    private String name;

}

File: android/app/src/main/java/com/example/richstonecargo/data/remote/dto/TopicDto.kt
package com.example.richstonecargo.data.remote.dto

import com.example.richstonecargo.domain.model.Topic

data class TopicDto(
    val id: Long,
    val name: String
)

fun TopicDto.toTopic(): Topic {
    return Topic(
        id = this.id,
        name = this.name
    )
}
